### PR TITLE
Pipeline threshold cleanup

### PR DESCRIPTION
**Pull request by cbebry**

### CODE REVIEW

File: python-project/mortar/image/threshold.py [hunk 1]
```diff
@@ -16,25 +15,17 @@
 class Threshold(Filter):
     """
     Perform thresholding on an image.
-
-    TODO this class doesn't yet support the necessary thresholding options.
     """
     name = 'Threshold'
 
-    def run(self, input: PILImage) -> Optional[PILImage]:
+    def run(self,
```
Comment (fmahnke) [inline python-project/mortar/image/threshold.py:21]: The pipeline composes any number of Filter objects and doesn't know their derived types. It calls the `run` method with the signature in the base `Filter` class, so this won't work there; the arguments all get their default values. See the `pipeline.py`.

Can you move the arguments to the constructor, store their values in some class attributes, and then use those values from the `run` method? See the `Crop` filter for an existing example of parametrization.

File: python-project/mortar/image/threshold.py [hunk 2]
```diff
@@ -16,25 +15,17 @@
...
-        for i in range(6):
-            plt.subplot(2, 3, i + 1)
-            plt.imshow(images[i], 'gray', vmin=0, vmax=255)
-            plt.title(titles[i])
-            plt.xticks([]), plt.yticks([])
-
-        return PIL.Image.frombytes('L', input.size, bytes(thresh1))
+
+        thresh_type = cv.THRESH_BINARY_INV if invert else cv.THRESH_BINARY
+        ret, thresh = cv.threshold(img, threshval, maxval, thresh_type)
```
Comment (fmahnke) [inline python-project/mortar/image/threshold.py:29]: `ret` would be the computed thresholding value. According to the OpenCV docs for this function, that value isn't even calculated for the binary/binary inverted thresholding types, so `ret` isn't something we'll use.

It's idiomatic to name "throwaway" variables using a single underscore, so I'd recommend doing that with `ret` in this case. See https://stackoverflow.com/questions/5893163/what-is-the-purpose-of-the-single-underscore-variable-in-python

File: python-project/mortar/image/threshold.py [hunk 3]
```diff
@@ -16,25 +15,17 @@
...
     """
     Perform thresholding on an image.
-
-    TODO this class doesn't yet support the necessary thresholding options.
     """
     name = 'Threshold'
 
-    def run(self, input: PILImage) -> Optional[PILImage]:
+    def run(self,
+            input: PILImage,
```
Comment (fmahnke) [inline python-project/mortar/image/threshold.py:22]: I was testing the updated filter in the pipeline and noticed that it doesn't seem to work correctly with RGB input images. It does process and return a new image, but the result isn't as expected, so I think we need to treat input images in non-RGB colorspace as an error condition.

Can you add an input check that makes this method return `None` if the image isn't grayscale? See the `Image.mode` attribute in the pillow library docs.

File: python-project/mortar/tesseract.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 """
-This module provides an interface for performing OCR operations using Tesseract.
```
Comment (fmahnke) [inline python-project/mortar/tesseract.py:2]: What's the reason for this formatting change?

File: python-project/test/test_image.py [hunk 5]
```diff
@@ -50,6 +50,14 @@ def test_threshold() -> None:
         output.save(Path(temp, 'image.png'))
 
 
+def test_threshold_rgb_rejection() -> None:
+    imageRGB = Image.new('RGB', (300, 200), (228, 150, 150))
+    threshold = Threshold()
+    outputRGB = threshold.run(imageRGB)
+    print(outputRGB)
```
Comment (fmahnke) [inline python-project/test/test_image.py:57]: Is there a reason you want to print this in the test output?

Comment (cbebry) [reply python-project/test/test_image.py]: Removed, mistake

File: python-project/mortar/tesseract.py (hunk 4 above)
Comment (cbebry) [reply python-project/mortar/tesseract.py:2]: 1 character too long, failed flake8 linting

**Review by fmahnke — APPROVED** [review]

**Pull request merged**